apps/studio/src/features/editing-experience/components/ComplexEditorStateDrawer.tsx

```diff
@@ -104,6 +111,59 @@ export default function ComplexEditorStateDrawer(): JSX.Element {
     setPreviewPageState(newPageState)
   }
 
+  const handleSave = async () => {
+    let newPageState = previewPageState
+
+    if (modifiedAssets.length > 0) {
+      const updatedBlocks = Array.from(previewPageState.content)
+      const newBlock = _.cloneDeep(updatedBlocks[currActiveIdx])
+
+      if (!newBlock) {
+        return
+      }
+
+      // Upload all new/modified images/files
+      const assetsToUpload = modifiedAssets.filter((asset) => !!asset.file)
+      await Promise.all(
```

Comment:
What happens if one of the upload fails?

Reply:
All the uploads should fail and the not found placeholder image will be shown. It is done this way because only when everything is successful, then we will update the block in the database.

Reply:
you'll need a try catch here i think

Reply:
Added the try-catch + toast message in 89b2d6c9da1c8538c8fd9068966c81970308a10a.